src/printer.js

```diff
@@ -2051,3 +2104,21 @@ function isLastStatement(path) {
   const body = parent.body;
   return body && body[body.length - 1] === node;
 }
+
+function collectLogicalExpressions(node, input=[]) {
```

Comment:
There's probably a higher order function that could be formulated here, but I'll probably rewrite this following the template in `printMemberChain` which seems to face similar issues